src/main/java/core/basesyntax/service/ShopServiceImpl.java

```
@@ -0,0 +1,24 @@
+package core.basesyntax.service;
+
+
+import java.util.List;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationHandler;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.get(transaction.getOperation());
+            if (handler != null) {
+                handler.handle(transaction);
```

If `operationStrategy.get(...)` returns null the current code silently skips the transaction (`if (handler != null) { ... }`). The requirements ask to handle incorrect/unknown operation/strategy mapping (do not silently ignore): "Incorrect data in the input file, for example, quantity less than zero or incorrect strategy". Instead of skipping, throw a RuntimeException with an informative message (include the transaction or its operation) so the caller/test can see the failure. Alternatively, make `OperationStrategy.get(...)` throw when a mapping is absent and let that propagate with a clear message.